crates/bevy_tasks/src/task_pool.rs

```diff
@@ -377,23 +382,29 @@ impl TaskPool {
 
                 // we get this from a thread local so we should always be on the scope executors thread.
                 let scope_ticker = scope_executor.ticker().unwrap();
-                if let Some(external_ticker) = external_executor.ticker() {
-                    if tick_task_pool_executor {
+                match (external_executor.ticker(), tick_task_pool_executor) {
+                    (Some(external_ticker), true)
+                        if !external_ticker.is_same_executor(&scope_ticker) =>
+                    {
                         Self::execute_global_external_scope(
                             executor,
                             external_ticker,
                             scope_ticker,
                             get_results,
                         )
                         .await
-                    } else {
+                    }
+                    (Some(external_ticker), false)
+                        if !external_ticker.is_same_executor(&scope_ticker) =>
+                    {
                         Self::execute_external_scope(external_ticker, scope_ticker, get_results)
                             .await
                     }
-                } else if tick_task_pool_executor {
-                    Self::execute_global_scope(executor, scope_ticker, get_results).await
-                } else {
-                    Self::execute_scope(scope_ticker, get_results).await
+                    // either external_executor is none or it is same as scope_executor
+                    (_, true) => {
```

Comment:
There's only one other case here and that's `None`, no need to squash that.

Reply:
This is also `(Some(_), true) if is_same_executor =>`

Reply:
> There's only one other case here and that's `None`, no need to squash that.

I put a comment above, none or is_same_executor both in this branch.

Reply:
Sorry if I was unclear, I'd rather this be explicit: rely on Rust's exhaustiveness checking and use `(None, true)` here instead of `_`, same with the case below.

Reply:
it covers more than just the (None, true) case. see my above comment.